core/src/main/java/com/scalar/db/api/Operation.java

```diff
@@ -191,6 +197,25 @@ public Operation withConsistency(Consistency consistency) {
     return this;
   }
 
+  /**
+   * Returns the attributes for this operation.
+   *
+   * @return the attributes
+   */
+  public Map<String, String> getAttributes() {
```

Comment:
[minor] I think this can return `ImmutableMap<>` to make it easy to construct Immutable objects without copying on the caller side.

Reply:
Since `getAttributes()` is a user-facing API, I didn’t expose the implementation of the `Map` interface here. I usually use immutable collections like `ImmutableMap` for internal APIs, but I avoid using them in exposed APIs. What do you think?

Reply:
> Since getAttributes() is a user-facing API

Oh, I see. Agreed. Let's go with returning `Map` as is.